src/main/java/EventsAndCommands/CommandLoader.java

```diff
@@ -0,0 +1,222 @@
+        loadedCommands = new ArrayList<>();
+        loadedListeners = new ArrayList<>();
+        setDb(db);
+        setConfig(config);
+        setWaiter(waiter);
+    }
+
+    public void loadCommands() {
+        Reflections reflections = new Reflections(this.getClass().getPackageName());
+
+        Set<Class<? extends Command>> cmds = reflections.getSubTypesOf(Command.class);
+        cmds = cmds.stream()
+            .filter(x -> x.getAnnotation(IgnoreCommand.class) == null)
+            .collect(Collectors.toSet());
+
+        List<Command> instances = new ArrayList<>();
+
+        for (var cmd : cmds) {
+            List<Constructor<?>> ctors = Arrays.asList(cmd.getConstructors());
+            for (var ctor : ctors) {
+                if (hasAcceptableParams(ctor)) {
+                    try {
+                        instances.add(injectCommand(ctor));
+                    } catch (InvocationTargetException e){
+                        e.getCause().printStackTrace();
+                    } catch (Exception e) {
+                        // TODO: Log
+                        System.err.println(e.getMessage());
+                        System.err.println("Could not load " + cmd.getName());
+                    }
+                } else {
+                    System.err.println("No suitable constructor for " + cmd.getName());
+                }
+            }
+        }
+
+        setLoadedCommands(instances);
+    }
+
+    public void loadListenerAdaptor() {
+        Reflections reflections = new Reflections(this.getClass().getPackageName());
+
+        Set<Class<? extends ListenerAdapter>> cmds = reflections.getSubTypesOf(ListenerAdapter.class);
+        cmds = cmds.stream()
+            .filter(x -> x.getAnnotation(IgnoreCommand.class) == null)
+            .collect(Collectors.toSet());
+            
+        List<ListenerAdapter> instances = new ArrayList<>();
+
+        for (var cmd : cmds) {
+            List<Constructor<?>> ctors = Arrays.asList(cmd.getConstructors());
+            for (var ctor : ctors) {
+                if (hasAcceptableParams(ctor)) {
+                    try {
+                        instances.add(injectListenerAdaptor(ctor));
+                    } catch (InvocationTargetException e){
+                        e.getCause().printStackTrace();
+                    } catch (Exception e) {
+                        // TODO: Log
+                        System.err.println(e.getMessage());
+                        System.err.println("Could not load " + cmd.getName());
+                    }
+                } else {
+                    System.err.println("No suitable constructor for " + cmd.getName());
+                }
+            }
+        }
+
+        setLoadedListeners(instances);
+    }
+
+    /**
+     * 
+     * @param ctors
+     * @return
+     */
+    public ListenerAdapter injectListenerAdaptor(Constructor<?> ctor) throws Exception{
```

Comment:
Expect for the final cast this looks identical to injectCommand? I'd use one more generic method, maybe casting to a type token class you pass it.